### PR TITLE
Refactor insight collector

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/ops/insightcollector/deployments.go [hunk 1]
```diff
@@ -451,3 +337,15 @@ func extractChangeFailureRate(deployments []*model.Deployment, from, to int64, t
 		FailureCount: failureCount,
 	}, rest
 }
+
+// groupDeployments groups deployments by applicationID and projectID
+func groupDeployments(deployments []*model.Deployment) (apps, projects map[string][]*model.Deployment) {
+	apps = make(map[string][]*model.Deployment)
+	projects = make(map[string][]*model.Deployment)
```
Comment (khanhtc1202) [inline pkg/app/ops/insightcollector/deployments.go:344]: nits
```golang
	apps = make(map[string][]*model.Deployment, len(deployments))
	projects = make(map[string][]*model.Deployment, len(deployments))
```

File: pkg/app/ops/insightcollector/deployments.go [hunk 2]
```diff
@@ -30,129 +27,131 @@ import (
...
+
+		deployments = append(deployments, d...)
+		maxCreatedAt = d[len(d)-1].CreatedAt
 	}
+	return deployments, nil
+}
 
-	return updateErr
+func (c *InsightCollector) findDeploymentsCompletedInRange(ctx context.Context, from, to int64) ([]*model.Deployment, error) {
+	const pageSize = 50
```
Comment (khanhtc1202) [inline pkg/app/ops/insightcollector/deployments.go:112]: should we keep it as global var/const as previously 🤔

File: pkg/app/ops/insightcollector/deployments.go (hunk 1 above)
Comment (khanhtc1202) [reply pkg/app/ops/insightcollector/deployments.go:344]: ah, my bad, my suggestion was wrong 🙏

File: pkg/app/ops/insightcollector/deployments.go [hunk 3]
```diff
@@ -451,3 +337,15 @@ func extractChangeFailureRate(deployments []*model.Deployment, from, to int64, t
 		FailureCount: failureCount,
 	}, rest
 }
+
+// groupDeployments groups deployments by applicationID and projectID
+func groupDeployments(deployments []*model.Deployment) (apps, projects map[string][]*model.Deployment) {
```
Comment (khanhtc1202) [inline pkg/app/ops/insightcollector/deployments.go:342]: should we add tests for this function 👀

Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go:342]: Sure.

File: pkg/app/ops/insightcollector/deployments.go (hunk 2 above)
Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go:112]: I have no strong opinion about this. Let me revert it.

File: pkg/app/ops/insightcollector/deployments.go (hunk 3 above)
Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go:342]: Actually, just found a bug from this function. 👍

Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go:342]: Added and fixed!

File: pkg/app/ops/insightcollector/deployments.go [hunk 4]
```diff
@@ -30,129 +27,131 @@ import (
...
-			i.logger.Error("failed to update application chunks", zap.Error(err))
+		if err := c.updateApplicationChunks(ctx, ds[0].ProjectId, id, ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
+			c.logger.Error("failed to update application chunks", zap.Error(err))
 			updateErr = err
 		}
 	}
 	for id, ds := range projects {
-		if err := i.updateApplicationChunks(ctx, id, ds[0].ApplicationId, ds, model.InsightMetricsKind_CHANGE_FAILURE_RATE, target); err != nil {
-			i.logger.Error("failed to update application chunks", zap.Error(err))
+		if err := c.updateApplicationChunks(ctx, id, "", ds, model.InsightMetricsKind_DEPLOYMENT_FREQUENCY, target); err != nil {
```
Comment (khanhtc1202) [inline pkg/app/ops/insightcollector/deployments.go:63]: After the update of `groupDeployments` logic, I finally got the point here 😂 
Just to confirm, is it okay to just pass an empty string here as `appID` value? 🤔

Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go:63]: Oh no. Don't know why this happened. Nice catch!

Comment (khanhtc1202) [reply pkg/app/ops/insightcollector/deployments.go:63]: looks like it buggy from start (before this refactor) 😂 Let's rethink the behaviors of this `updateApplicationChunks` method 👍

Comment (nghialv) [reply pkg/app/ops/insightcollector/deployments.go]: Fixed this change.

Let me keep this PR as its title, just doing the refactoring and keep its logic as before.
And I will check the logic later and make another PR if needed.

Comment (khanhtc1202) [reply pkg/app/ops/insightcollector/deployments.go]: 🙆‍♀️ Please make an issue for this later 🙏

**Comment by khanhtc1202** [comment]
🚀 
/lgtm

**Comment by nakabonne** [comment]
👍 
/approve

**Pull request merged**